pkg/controller/run/parse_line.go

```diff
@@ -214,48 +208,36 @@ func (c *Controller) parseSemverTagLine(ctx context.Context, logE *logrus.Entry,
 			return patchLine(action, sha, lv), nil
 		}
 	}
-	// verify commit hash
-	if !c.param.IsVerify {
-		return "", nil
-	}
-	// @xxx # v3.0.0
-	// @<full commit hash> # v3.0.0
-	if FullCommitSHA != getVersionType(action.Version) {
-		return "", nil
-	}
-	if err := c.verify(ctx, action); err != nil {
-		return "", fmt.Errorf("verify the version annotation: %w", err)
-	}
-	return "", nil
-}
 
-func (c *Controller) parseShortSemverTagLine(ctx context.Context, logE *logrus.Entry, action *Action) (string, error) {
-	// @xxx # v3
-	// @<full commit hash> # v3
+	// ensure the commit sha is valid
 	if FullCommitSHA != getVersionType(action.Version) {
 		return "", ErrCantPinned
 	}
-	if c.param.Update {
-		lv, err := c.getLatestVersion(ctx, logE, action.RepoOwner, action.RepoName)
-		if err != nil {
-			return "", fmt.Errorf("get the latest version: %w", err)
-		}
-		sha, _, err := c.repositoriesService.GetCommitSHA1(ctx, action.RepoOwner, action.RepoName, lv, "")
+
+	// if we have a "short version" (e.g. not a complete `1.2.3` semver), check
+	// for a tag that matches the commit hash. note that this could be the same
+	// tag that's specified
+	versionParts := regexp.MustCompile(`\d+`).FindAllString(action.VersionComment, -1)
+	// using != supports validation against version tags like:
+	// - v1 (1 part)
+	// - v1.2 (2 parts)
+	// - v1.2.3.4 (4 parts)
+	if len(versionParts) != 3 {
+		tag, err := c.getTagFromSHA(ctx, action, action.Version)
 		if err != nil {
-			return "", fmt.Errorf("get the latest version: %w", err)
+			return "", err
 		}
-		return patchLine(action, sha, lv), nil
-	}
-	// replace Shortsemer to Semver
-	longVersion, err := c.getLongVersionFromSHA(ctx, action, action.Version)
-	if err != nil {
-		return "", err
+
+		return patchLine(action, action.Version, tag), nil
 	}
-	if longVersion == "" {
-		logE.Warn("a long tag whose SHA is same as SHA of the version annotation isn't found")
+
+	if !c.param.IsVerify {
 		return "", nil
```

Comment:
this won't be called for non-semantic tags like `v4`, which i _think_ is fine, because the previous `parseShortSemverTagLine()` didn't support validating them.